drivers/led_strip/led_strip_shell.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2025 Dmitrii Sharshakov <d3dx12.xx@gmail.com>
+ *
+ * Based on drivers/led/led_shell.c which is:
+ * Copyright (c) 2020 Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/kernel.h"
+#include "zephyr/sys/util.h"
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/led_strip.h>
+#include <stdlib.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(led_shell);
+
+#define MAX_COLOR_ARGS 255
+
+enum {
+	arg_idx_dev	= 1,
+	arg_idx_value	= 2,
+};
+
+static int parse_common_args(const struct shell *sh, char **argv,
+			     const struct device **dev)
+{
+	*dev = shell_device_get_binding(argv[arg_idx_dev]);
+	if (!*dev) {
+		shell_error(sh,
+			    "LED device %s not found", argv[arg_idx_dev]);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int cmd_update_rgb(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	size_t num_colors, strip_len;
+	uint8_t i;
+	struct led_rgb *pixels;
+
+	err = parse_common_args(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	strip_len = led_strip_length(dev);
+	num_colors = argc - arg_idx_value;
+	if (num_colors > MIN(strip_len, MAX_COLOR_ARGS)) {
+		shell_error(sh,
+			"Invalid number of colors %d (max %d)",
+			num_colors, MIN(strip_len, MAX_COLOR_ARGS));
+		return -EINVAL;
+	}
+
+	pixels = k_malloc(num_colors * sizeof(struct led_rgb));
```

Comment:
Avoid using `k_malloc` altogether, or at least free the memory.

Reply:
Oops, a terrible mistake on my side. Should I just lower the number of LEDs supported and store it on stack?